datastore/datastore/storagebackend/postgresql/postgresql.go

```diff
@@ -21,6 +22,82 @@ func (sbe *PostgreSQL) Description() string {
 	return "PostgreSQL database"
 }
 
+// setTSUniqueMainCols extracts into tsStringMdataPBNamesUnique the columns comprising constraint
+// unique_main in table time_series.
+//
+// Returns nil upon success, otherwise error.
+func (sbe *PostgreSQL) setTSUniqueMainCols() error {
+
+	query := `
+		SELECT pg_get_constraintdef(c.oid)
+		FROM pg_constraint c
+		JOIN pg_namespace n ON n.oid = c.connamespace
+		WHERE conrelid::regclass::text = 'time_series'
+			AND conname = 'unique_main'
+			AND contype = 'u'
+	`
+
+	row := sbe.Db.QueryRow(query)
```

Comment:
Can you give an example of the query output?

Reply:
```
$ PGPASSWORD=mysecretpassword psql -h localhost -p 5433 -U postgres -d data -t -c "SELECT pg_get_constraintdef(c.oid) FROM pg_constraint c JOIN pg_namespace n ON n.oid = c.connamespace WHERE conrelid::regclass::text = 'time_series' AND conname = 'unique_main' AND contype = 'u'"
 UNIQUE NULLS NOT DISTINCT (naming_authority, platform, standard_name, instrument, level, period, function)
```

Reply:
I think it would help to have this in an example, as it helps to understand the code.

Reply:
Good point - will do.